### PR TITLE
Define C API for eigenvector centrality

### PR DESCRIPTION
**Pull request by ChuckHastings**
This PR defines the C API for eigenvector centrality.  It is the first of a series of PRs to address #2146

It also does the following:
* Renames `cugraph_pagerank_result_t` to `cugraph_centrality_result_t` to allow it to be reused for all centrality algorithms
* Reorganizes the code for the results into a separate file
* Provides an initial implementation of eigenvector centrality that can be used for testing the API.  NOTE: It literally calls pagerank with some hardcoded parameters.  The return formats for the algorithms are the same, so this will mimic the API behavior (although the scores will be different once eigenvector centrality is implemented
* Provided a C unit test that validates the results (will need to be updated to validate eigenvector centrality results once the real algorithm is in place)

Note that this is a breaking change since it modifies the type of the pagerank result.

### CODE REVIEW

File: cpp/src/c_api/centrality_result.hpp
Comment (seunghwak) [inline cpp/src/c_api/centrality_result.hpp:26]: In line with the previous comment, centrality_values_ (or just values_ as centrality appears in the struct name) may sound a bit less ambiguous?

File: cpp/src/c_api/eigenvector_centrality.cpp
Comment (seunghwak) [inline cpp/src/c_api/eigenvector_centrality.cpp:38]: Better add `{}` to avoid holding garbage values?

File: cpp/src/c_api/centrality_result.hpp
Comment (seunghwak) [reply cpp/src/c_api/centrality_result.hpp:26]: And to add '{}' to avoid holding garbage values.

File: cpp/src/c_api/eigenvector_centrality.cpp
Comment (seunghwak) [inline cpp/src/c_api/eigenvector_centrality.cpp:62]: Should we do anything different? (compared to PageRank & Katz centrality?)

File: cpp/src/c_api/pagerank.cpp
Comment (seunghwak) [inline cpp/src/c_api/pagerank.cpp:39]: Add `{}`

**Review by seunghwak — COMMENTED** [review]
Looks good to me except for few nitpicking comments.

File: cpp/include/cugraph_c/algorithms.h
Comment (ChuckHastings) [reply cpp/include/cugraph_c/algorithms.h:49]: Fixed in next push

File: cpp/src/c_api/eigenvector_centrality.cpp
Comment (ChuckHastings) [reply cpp/src/c_api/eigenvector_centrality.cpp:38]: Fixed in next push

File: cpp/src/c_api/eigenvector_centrality.cpp
Comment (ChuckHastings) [reply cpp/src/c_api/eigenvector_centrality.cpp:62]: Regarding the FIXME?

I should probably create an issue and delete all of the FIXMEs that are propagating here.  The issue is to come up with a way to have an SG only version of cugraph.  Some customers don't need an MG version and an SG only package of cugraph would be lighter weight for them to install.  But this aspect of the C API implementation would need to be addressed somehow.

However it is addressed, it would be addressed the same in each API.

File: cpp/src/c_api/pagerank.cpp
Comment (ChuckHastings) [reply cpp/src/c_api/pagerank.cpp:39]: Done in next push

**Review by seunghwak — APPROVED** [review]

**Review by rlratzel — APPROVED** [review]
Just some minor observations which need not hold up approval.

File: cpp/src/c_api/centrality_result.cpp
Comment (ChuckHastings) [reply cpp/src/c_api/centrality_result.cpp]: Yes.  Pushed update.

File: cpp/src/c_api/eigenvector_centrality.cpp
Comment (ChuckHastings) [reply cpp/src/c_api/eigenvector_centrality.cpp]: Yes.  Pushed update.

**Review by rlratzel — APPROVED** [review]

**Comment by ChuckHastings** [comment]
@gpucibot merge

**Pull request merged**
